packages/next/src/build/webpack/plugins/next-trace-entrypoints-plugin.ts

```diff
@@ -260,22 +289,60 @@ export class TraceEntryPointsPlugin implements webpack.WebpackPluginInstance {
         }
 
         const finalFiles: string[] = []
-
-        for (const file of new Set([
-          ...entryFiles,
-          ...(this.entryTraces.get(entrypoint.name) || []),
-        ])) {
-          if (file) {
-            finalFiles.push(
-              nodePath.relative(traceOutputPath, file).replace(/\\/g, '/')
-            )
-          }
+        const readFile = (path: string): Promise<Buffer | string> => {
+          return new Promise((resolve, reject) => {
+            compilation.inputFileSystem.readFile(path, (err, result) => {
+              if (err) {
+                return reject(err)
+              }
+              resolve(result || '')
+            })
+          })
         }
+        const fileHashes: Record<string, string> = {}
+
+        await Promise.all(
```

Comment:
should we skip this work if `flyingShuttle` is disabled or is there still a use for `fileHashes` outside of it?

Reply:
We do skip the hashing part if `flyingShuttle` is disabled we still do the `finalFiles` collecting though